R/aorsf_data.R

```diff
@@ -0,0 +1,225 @@
+          n_thread = 1,
+          verbose_progress = FALSE
+        )
+    )
+  )
+
+  parsnip::set_encoding(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "classification",
+    options = list(
+      predictor_indicators = "none",
+      compute_intercept = FALSE,
+      remove_intercept = FALSE,
+      allow_sparse_x = FALSE
+    )
+  )
+
+  parsnip::set_fit(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "regression",
+    value = list(
+      interface = "formula",
+      protect = c("formula", "data", "weights"),
+      func = c(pkg = "aorsf", fun = "orsf"),
+      defaults =
+        list(
+          n_thread = 1,
+          verbose_progress = FALSE
+        )
+    )
+  )
+
+  parsnip::set_encoding(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "regression",
+    options = list(
+      predictor_indicators = "none",
+      compute_intercept = FALSE,
+      remove_intercept = FALSE,
+      allow_sparse_x = FALSE
+    )
+  )
+
+
+  parsnip::set_pred(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "classification",
+    type = "class",
+    value = list(
+      pre = NULL,
+      # makes prob preds consistent with class ones
+      post = function(results, object){
+
+        missings <- apply(results, 1, function(x) any(is.na(x)))
+
+        if(!any(missings))
+          return(colnames(results)[apply(results, 1, which.max)])
+
+        obs <- which(!missings)
+
+        out <- rep(NA_character_, nrow(results))
+        out[obs] <- colnames(results)[apply(results[obs, ], 1, which.max)]
+        out
+
+      },
+      func = c(fun = "predict"),
+      args =
+        list(
+          object = quote(object$fit),
+          new_data = quote(new_data),
+          pred_type = "prob",
+          verbose_progress = FALSE,
+          na_action = 'pass'
+        )
+    )
+  )
+
+  parsnip::set_pred(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "classification",
+    type = "prob",
+    value = list(
+      pre = NULL,
+      post = function(x, object) {
+        as_tibble(x)
+      },
+      func = c(fun = "predict"),
+      args =
+        list(
+          object = quote(object$fit),
+          new_data = quote(new_data),
+          pred_type = 'prob',
+          verbose_progress = FALSE,
+          na_action = 'pass'
+        )
+    )
+  )
+
+  parsnip::set_pred(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "classification",
+    type = "raw",
+    value = list(
+      pre = NULL,
+      post = NULL,
+      func = c(fun = "predict"),
+      args =
+        list(
+          object = quote(object$fit),
+          new_data = quote(new_data),
+          verbose_progress = FALSE,
+          na_action = 'pass'
+        )
+    )
+  )
+
+  parsnip::set_pred(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "regression",
+    type = "numeric",
+    value = list(
+      pre = NULL,
+      post = function(results, object)
+        as.numeric(results),
```

Comment:
```suggestion
      post = as.numeric
```

Or just NULL if parsnip knows to convert to numeric in this case. Would guess that `as.numeric()`'s dots are quietly ignored here but ignore if not.

Reply:
Same comment would go for the line below.